woocommerce/wc-admin
# Basic customer effort score survey functionality

This is a spike of the customer effort score feature described in https://github.com/woocommerce/woocommerce-admin/issues/5259.

At the moment the modal looks like this:
![image](https://user-images.githubusercontent.com/224531/95692063-498bde00-0c67-11eb-98c4-64bffd467331.png)

It is behind a feature toggle, enabled for development only. This consists of a component in `@woocommerce/customer-effort-score`, which is the modal itself which accepts a callback for submitting the survey. This is wrapped by `client/customer-effort-score-tracks/index.js`, which uses the Tracks library for tracking. This wrapping lets the survey be independent of the data collection method, allowing reuse between teams.

### Steps to test:

1. Add a new product. The modal should be displayed on the edit product page (the page you are taken to after publishing a new product).
2. Click the "Click me" button. Tracks data should be outputted, eg:

```
wcadmin_product_add_publish_effort_score,
{
    score: 0,
    product_count: 146
}
```
3. Test this with editing a product as well

Note that the event name and properties are going to change in subsequent PRs.
